sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -74,6 +76,15 @@ case class CreateFunctionCommand(
   }
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
+    // Checks if the given resources exist
+    val hadoopConf = sparkSession.sparkContext.hadoopConfiguration
+    val nonExistentResources = resources.filter { r =>
+      val path = new Path(r.uri)
```

Comment:
it will also break one case with local schema

```
scala> new Path("local:/path").getFileSystem(spark.sparkContext.hadoopConfiguration)
java.io.IOException: No FileSystem for scheme: local
  at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2660)
  at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2667)
  at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:94)
  at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2703)
  at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2685)
  at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
  at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
  ... 47 elided
```

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L1903

I think I rushed to review this one. Reverting this.

Reply:
```
scala> sql("""CREATE FUNCTION udtf_test AS 'org.apache.spark.sql.hive.execution.UDFTest' USING JAR 'local:/var/invalid/invalid.jar'
     | """)
java.io.IOException: No FileSystem for scheme: local
  at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2660)
  at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2667)
  at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:94)
  at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2703)
  at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2685)
```

Reply:
okay soon will raise the follow-up PR with all the test coverage

Reply:
At present, Spark does not check whether this class is in the jar when registering a permanent UDF.
But hive does check if jar and class exist.
Can we also add a check for the existence of class in spark, like registering a temporary function?

https://github.com/apache/hive/blob/9f40d7cc1d889aa3079f3f494cf810fabe326e44/ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java#L289-L296

https://github.com/apache/hive/blob/9f40d7cc1d889aa3079f3f494cf810fabe326e44/ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java#L686-L691